src/amo/components/AddonReviewCard/index.js

```diff
@@ -380,29 +371,45 @@ export class AddonReviewCardBase extends React.Component<InternalProps> {
         );
       }
 
+      const byLineLink = replaceStringsWithJSX({
+        text: i18n.sprintf(byLineString, {
+          authorName: review.userName,
+          timestamp: i18n.moment(review.created).fromNow(),
+          // Keep the link placeholders so that we can use them to inject a
+          // `<Link />` using `replaceStringsWithJSX`.
```

Comment:
Aww, man, I didn't think about how other format strings need to co-exist. This is confusing. One solution would be to allow simple replacements, like:

```diff
diff --git a/src/amo/components/AddonReviewCard/index.js b/src/amo/components/AddonReviewCard/index.js
index bc3fdd31f..7b429a581 100644
--- a/src/amo/components/AddonReviewCard/index.js
+++ b/src/amo/components/AddonReviewCard/index.js
@@ -372,15 +372,10 @@ export class AddonReviewCardBase extends React.Component<InternalProps> {
       }
 
       const byLineLink = replaceStringsWithJSX({
-        text: i18n.sprintf(byLineString, {
-          authorName: review.userName,
-          timestamp: i18n.moment(review.created).fromNow(),
-          // Keep the link placeholders so that we can use them to inject a
-          // `<Link />` using `replaceStringsWithJSX`.
-          linkEnd: '%(linkEnd)s',
-          linkStart: '%(linkStart)s',
-        }),
+        text: byLineString,
         replacements: [
+          ['authorName', () => review.userName],
+          ['timestamp', () => i18n.moment(review.created).fromNow()],
           [
             'linkStart',
             'linkEnd',
```

However, this will make things complicated so maybe it's not worth it. As long as you have a comment explaining the double format strings, it's not so bad.